sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -376,16 +389,11 @@ public void finalizeCheckpoint() throws IOException {
       this.source = source;
       this.current = null;
       this.checkpointMark = checkpointMark != null ? checkpointMark : new RabbitMQCheckpointMark();
-      try {
-        connectionHandler = new ConnectionHandler(source.spec.uri());
```

Comment:
Fixing a separate bug here. Originally, the constructor created the `connectionHandler` instance, which was closed inside `close()`. However `start()` created one as well, but masked the member variable with a local instance, so the channel/connection actually used for processing was not the one managed by the lifecycle of the class.

This is a semantic difference though, as now validation of the connection handler happens during `start()` rather than during construction. If there's a strong preference, I can restore this and instead modify `start()` to use the existing connection handler

Reply:
My impression is that your change here is an improvement.